src/mito2/src/memtable/time_series.rs

```diff
@@ -339,29 +360,112 @@ struct Values {
 }
 
 impl Values {
+    pub fn new_empty(metadata: &RegionMetadataRef) -> Self {
+        let timestamp = metadata
+            .schema
+            .timestamp_column()
+            .unwrap()
+            .data_type
+            .create_mutable_vector(0)
+            .to_vector();
+        let sequence = Arc::new(UInt64Vector::from_vec(vec![]));
+        let op_type = Arc::new(UInt8Vector::from_vec(vec![]));
+        let fields = metadata
+            .field_columns()
+            .map(|f| {
+                f.column_schema
+                    .data_type
+                    .create_mutable_vector(0)
+                    .to_vector()
+            })
+            .collect();
+        Self {
+            timestamp,
+            sequence,
+            op_type,
+            fields,
+        }
+    }
+
     /// Converts [Values] to `Batch`, sorts the batch according to `timestamp, sequence` desc and
     /// keeps only the latest row for the same timestamp.
-    pub fn to_batch(&self, primary_key: &[u8], metadata: &RegionMetadataRef) -> Result<Batch> {
+    pub fn to_batch(
+        &self,
+        primary_key: &[u8],
+        metadata: &RegionMetadataRef,
+        projection: Option<&[ColumnId]>,
```

Comment:
Could we just store the projection as a `HashSet<ColumnId>` and check whether we need a column by searching the hash set?

Reply:
I think we should preserve the order of field columns as in projection.